pkg/gatherers/clusterconfig/gather_cluster_ingress_certificates.go

```diff
@@ -0,0 +1,239 @@
+// It also collects the name and namespace of any Ingress Controllers using the certificates.
+//
+// ### API Reference
+// - https://docs.openshift.com/container-platform/4.13/rest_api/operator_apis/ingresscontroller-operator-openshift-io-v1.html
+// - https://docs.openshift.com/container-platform/4.13/rest_api/security_apis/secret-v1.html
+//
+// ### Sample data
+// - docs/insights-archive-sample/aggregated/ingress_controllers_certs.json
+//
+// ### Location in archive
+// - `aggregated/ingress_controllers_certs.json`
+//
+// ### Config ID
+// `clusterconfig/ingress_certificates`
+//
+// ### Released version
+// - 4.17
+//
+// ### Backported versions
+// TBD
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherClusterIngressCertificates(ctx context.Context) ([]record.Record, []error) {
+	const Filename = "aggregated/ingress_controllers_certs"
+
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	operatorClient, err := operatorclient.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	certificates, errs := gatherClusterIngressCertificates(ctx, gatherKubeClient.CoreV1(), operatorClient)
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	return []record.Record{{
+		Name: Filename,
+		Item: record.JSONMarshaller{Object: certificates},
+	}}, nil
+}
+
+func gatherClusterIngressCertificates(
+	ctx context.Context,
+	coreClient corev1client.CoreV1Interface,
+	operatorClient operatorclient.Interface) ([]*CertificateInfo, []error) {
+	//
+	var ingressAllowedNS = [2]string{"openshift-ingress-operator", "openshift-ingress"}
+
+	var certificatesInfo = make(map[CertificateInfoKey]*CertificateInfo)
+	var errs []error
+
+	// Step 1: Collect router-ca and router-certs-default
+	rCAinfo, err := getRouterCACertInfo(ctx, coreClient)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if rCAinfo != nil {
+		certificatesInfo[infereKey(rCAinfo)] = rCAinfo
+	}
+
+	rCDinfo, err := getRouterCertsDefaultCertInfo(ctx, coreClient)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	if rCDinfo != nil {
+		certificatesInfo[infereKey(rCDinfo)] = rCDinfo
+	}
+
+	// Step 2: List all Ingress Controllers
+	for _, namespace := range ingressAllowedNS {
+		controllers, err := operatorClient.OperatorV1().IngressControllers(namespace).List(ctx, metav1.ListOptions{})
+		if errors.IsNotFound(err) {
+			klog.V(2).Infof("Ingress Controllers not found in '%s' namespace", namespace)
+			continue
+		}
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		// Step 3: Filter Ingress Controllers with spec.defaultCertificate and get certificate info
+		for i := range controllers.Items {
+			controller := controllers.Items[i]
+
+			// Step 4: Check the certificate limits
+			if len(certificatesInfo) >= ingressCertificatesLimits {
+				klog.V(2).Infof("Reached the limit of ingress certificates (%d), skipping additional certificates", ingressCertificatesLimits)
```

Comment:
this message should be returned as an error so that it's available (visible) in the archive please.

Reply:
Solved in 0c0ab88f4fead85c740764bd8d36b31229b4ddb5